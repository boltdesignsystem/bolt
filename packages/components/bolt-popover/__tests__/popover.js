import { render, stopServer } from '../../../testing/testing-helpers';
import schema from '../popover.schema';
const { placement, spacing, theme } = schema.properties;

describe('<bolt-popover> Component', () => {
  afterAll(async () => {
    await stopServer();
  }, 100);

  // Basic Usage
  test('basic usage', async () => {
    const results = await render('@bolt-components-popover/popover.twig', {
      trigger:
        '<button type="button" class="e-bolt-button">This triggers a popover</button>',
      content:
        'This is the content of the popover with a <a href="https://pega.com" class="e-bolt-text-link">call to action</a>.',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // Props
  placement.enum.forEach(async placementChoice => {
    test(`content placement: ${placementChoice}`, async () => {
      const results = await render('@bolt-components-popover/popover.twig', {
        trigger:
          '<button type="button" class="e-bolt-button">This triggers a popover</button>',
        content:
          'This is the content of the popover with a <a href="https://pega.com" class="e-bolt-text-link">call to action</a>.',
        placement: placementChoice,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  spacing.enum.forEach(async spacingChoice => {
    test(`content spacing: ${spacingChoice}`, async () => {
      const results = await render('@bolt-components-popover/popover.twig', {
        trigger:
          '<button type="button" class="e-bolt-button">This triggers a popover</button>',
        content:
          'This is the content of the popover with a <a href="https://pega.com" class="e-bolt-text-link">call to action</a>.',
        spacing: spacingChoice,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  theme.enum.forEach(async themeChoice => {
    test(`content theme: ${themeChoice}`, async () => {
      const results = await render('@bolt-components-popover/popover.twig', {
        trigger:
          '<button type="button" class="e-bolt-button">This triggers a popover</button>',
        content:
          'This is the content of the popover with a <a href="https://pega.com" class="e-bolt-text-link">call to action</a>.',
        theme: themeChoice,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  test(`UUID of the popover`, async () => {
    const results = await render('@bolt-components-popover/popover.twig', {
      trigger:
        '<button type="button" class="e-bolt-button">This triggers a popover</button>',
      content:
        'This is the content of the popover with a <a href="https://pega.com" class="e-bolt-text-link">call to action</a>.',
      uuid: 'custom-unique-id',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
