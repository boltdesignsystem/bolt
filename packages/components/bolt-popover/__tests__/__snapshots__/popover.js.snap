// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-popover> Component UUID of the popover 1`] = `
<bolt-popover uuid="custom-unique-id">
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component basic usage 1`] = `
<bolt-popover uuid="12345">
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: auto 1`] = `
<bolt-popover placement="auto"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: bottom 1`] = `
<bolt-popover placement="bottom"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: bottom-end 1`] = `
<bolt-popover placement="bottom-end"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: bottom-start 1`] = `
<bolt-popover placement="bottom-start"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: left 1`] = `
<bolt-popover placement="left"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: left-end 1`] = `
<bolt-popover placement="left-end"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: left-start 1`] = `
<bolt-popover placement="left-start"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: right 1`] = `
<bolt-popover placement="right"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: right-end 1`] = `
<bolt-popover placement="right-end"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: right-start 1`] = `
<bolt-popover placement="right-start"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: top 1`] = `
<bolt-popover placement="top"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: top-end 1`] = `
<bolt-popover placement="top-end"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: top-start 1`] = `
<bolt-popover placement="top-start"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content spacing: medium 1`] = `
<bolt-popover spacing="medium"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content spacing: none 1`] = `
<bolt-popover spacing="none"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content spacing: small 1`] = `
<bolt-popover spacing="small"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content spacing: xsmall 1`] = `
<bolt-popover spacing="xsmall"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content theme: dark 1`] = `
<bolt-popover theme="dark"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content theme: light 1`] = `
<bolt-popover theme="light"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content theme: none 1`] = `
<bolt-popover theme="none"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content theme: xdark 1`] = `
<bolt-popover theme="xdark"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;

exports[`<bolt-popover> Component content theme: xlight 1`] = `
<bolt-popover theme="xlight"
              uuid="12345"
>
  <button type="button"
          class="e-bolt-button"
  >
    This triggers a popover
  </button>
  <span slot="content">
    This is the content of the popover with a
    <a href="https://pega.com"
       class="e-bolt-text-link"
    >
      call to action
    </a>
    .
  </span>
</bolt-popover>
`;
